tests/testdata/provider/testdata_pg.sh

```diff
@@ -27,57 +28,60 @@ SCRIPTS12="
 "
 
 echo "Dropping DB $DB"
-psql -q --echo-errors 'service=qgis_test dbname=postgres' -c "DROP DATABASE IF EXISTS $DB"
-
+dropdb --if-exists "${DB}"
 echo "Creating DB $DB"
-psql -q --echo-errors 'service=qgis_test dbname=postgres' -c "CREATE DATABASE $DB WITH TEMPLATE = template0 ENCODING = UTF8"  || exit 1
+createdb "${DB}" -E UTF8 -T template0 || exit 1
+
+export PGDATABASE="${DB}"
 
 for f in ${SCRIPTS}; do
   echo "Loading $f"
-  psql -q --echo-errors 'service=qgis_test' -c "SET client_min_messages TO WARNING;" -f "${f}" -v ON_ERROR_STOP=1 || exit 1
+  psql -q --echo-errors -c "SET client_min_messages TO WARNING;" -f "${f}" -v ON_ERROR_STOP=1 || exit 1
 done
 
-PGSERVERVERSION=$(psql -XtA 'service=qgis_test' -c 'SHOW server_version_num')
+PGSERVERVERSION=$(psql -XtA -c 'SHOW server_version_num')
 if test "${PGSERVERVERSION}" -gt 120000; then
   for f in ${SCRIPTS12}; do
     echo "Loading $f"
-    psql -q --echo-errors 'service=qgis_test' -c "SET client_min_messages TO WARNING;" -f "${f}" -v ON_ERROR_STOP=1 || exit 1
+    psql -q --echo-errors -c "SET client_min_messages TO WARNING;" -f "${f}" -v ON_ERROR_STOP=1 || exit 1
   done
 fi
 
 FINGERPRINT="${DB}-$(date +%s)"
-echo "Storing fingerprint ${FINGERPRINT} in $DB"
-psql -q --echo-errors 'service=qgis_test' -v ON_ERROR_STOP=1 <<EOF || exit 1
+echo "Storing fingerprint '${FINGERPRINT}' in $DB"
+
+psql -q --echo-errors -v ON_ERROR_STOP=1 <<EOF || exit 1
 CREATE TABLE qgis_test.schema_info AS SELECT
   'fingerprint' var, '${FINGERPRINT}' val;
 EOF
 
 # Test service=qgis_test connects to the just-created database
-CHECK=$(psql -XtA 'service=qgis_test' -c "select val from qgis_test.schema_info where var='fingerprint'")
+echo "Checking if we can connect to ${DB} via service=qgis_test"
+CHECK=$(psql -wXtA 'service=qgis_test' -c "select val from qgis_test.schema_info where var='fingerprint'")
 if test "${CHECK}" != "${FINGERPRINT}"; then
   exec >&2
   echo "ERROR: Could not access the just created test database ${DB} via service=qgis_test"
-  echo "HINT: create a section like the following in ~/.pg_service.conf"
+  echo "HINT: setup a section like the following in ~/.pg_service.conf"
   cat <<EOF
 [qgis_test]
   host=localhost
   port=5432
   dbname=${DB}
-  user=USERNAME
-  password=PASSWORD
+  user=qgis_test_user
+  password=qgis_test_user_password
 EOF
   exit 1
 fi
 
-# TODO: Test service=qgis_test connects via a method which accepts
+# Test service=qgis_test connects via a method which accepts
 # username/password
-CHECK=$(psql -XtA 'service=qgis_test user=qgis_test_user password=qgis_test_user_password' -c "select version()")
+CHECK=$(psql -wXtA 'service=qgis_test user=qgis_test_unprivileged_user password=qgis_test_unprivileged_user_password' -c "select version()")
```

Comment:
Do we still need the `service=qgis_test` here?
```suggestion
CHECK=$(psql -wXtA 'user=qgis_test_unprivileged_user password=qgis_test_unprivileged_user_password' -c "select version()")
```

Reply:
The aim of that final check is exactly _checking_ that you are properly setup to use service=qgis_test, so yes.
See how it works by temporary renaming your ~/.pg_service.conf and re-running test_data.pg.sh

Expected behaviour: you are notified you need an entry in ~/.pg_service.conf and you are also given an example such entry. Does this satisfy your request to have it documented ? It is also documented in tests/README.md, since 2017 (according to git-blame)

Reply:
Now I see you mean in the second check, right. Yes we still need it because we expect the service to specify hostname, database name, port etc. What we are *additionally* checking is that the service _also_ allows for different username/password (assuming the service block uses "qgis_test_user" for connecting, we check if we are able to also connect as "qgis_test_unprivileged_user"). 

This may be a too paranoid test, but unless you are getting errors for perfectly fine situation I don't see why changing it. I've added it because I was not using "qgis_test_user" but my own unix user in the "qgis_test" service via unix socket and the test failed when trying to connect with explicit username/password  (pg_hba.conf directive).

The aim of this PR is to simplify the bootstrap for people who want to run tests, so if you run into  problems, from a fresh machine/vm/container please report those problems.

Reply:
My point what: to do the first db connection `dropdb --if-exists "${DB}"` (as some others cmd after this line) you need to have credentials to access the db. These credentials are not defined if you simply run this script `./testdata_pg.sh` and without credentials the script runs from error to errors. I added the `service=qgis_test` everywhere to match the README. 

Now if you remove the `service=qgis_test` this script could only be ran if it is ran as `postgres` user or if it has the right env vars set before its execution (`PGDATABASE`, `PGUSER`, `PGPASSWORD`, etc.). And the README is not really clear (again from my point of view). 

Maybe a command line example in the README could do the trick or add at the beginning of the script a check to validate the PG credentials?

`if ! psql -XtA -c 'SHOW server_version_num' >/dev/null; then echo "Set PG credentials!"; exit 1; fi`

Reply:
First of all I'll describe you my experience: I have ident based authentication for unix socket and my unix username matches a superuser on the cluster, so with default credentials I'm acting as superuser, and thus my running `testdata_pg.sh` just works always.

Now to your experience: `testdata_pg.sh` fails from error to error. My solution to that would be to exit earlier and give more tips about what it takes for it to run successfully.

What it takes for `testdata_pg.sh` to run successfully is that the environment, or psql configuration, makes is so that the running user is recognized as a cluster user with the privileged of creating databases and roles. I've pushed a commit adding this info, for now. Will see about having the .sh file perform pre-checks

Reply:
I've tested running the script as an underprivileged user and I got error pretty early and it was very clear. I pushed a new commit that exit even earlier if the existing qgis_test database cannot be dropped.

By testing it I notice it really takes SUPERUSER privileges (at the moment) due to the need to create PostGIS extension, which is non-trusted. I'm not sure about writing this need in the README because you _may_ be running pgextwlist extension which would let you create PostGIS extension even if you are not superuser. You woudl still need privileges to create databases and roles. I think the README is now clear about that.